packages/tracing/src/browser/router.ts

```diff
@@ -3,6 +3,63 @@ import { addInstrumentationHandler, getGlobalObject, logger } from '@sentry/util
 
 const global = getGlobalObject<Window>();
 
+const PATHNAME_PARAMETER_PATTERNS = {
+  number: /^\d+$/,
+  'sha1-hash': /^[0-9a-f]{40}$/i,
+  'md-hash': /^[0-9a-f]{32}$/i,
+  uuid: /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i,
+};
+
+/**
+ * Tries to parameterize a provided pathname based on some heuristics.
+ *
+ * @param pathname - A pathname, usually obtained via `window.location.pathname`.
+ * @returns a parameterized version of the pathname alongside the values behind the parameters
+ */
+function extractPathnameParametersWithHeuristics(pathname: string): {
+  parameterizedPathname: string;
+  pathnameParameterValues: Record<string, string>;
+  hasParameters: boolean;
+} {
+  /**
+   * Keeps track of the number of occurences of each pattern in the provided pathname.
+   */
+  const patternCounts: Partial<Record<keyof typeof PATHNAME_PARAMETER_PATTERNS, number>> = {};
+
+  /**
+   * Keeps track of what found parameters in the URL evaluate to, e.g:
+   * { "uuid-1": "2778216e-b40c-46...", "uuid-2": "86024957-1789-47ec-be7..." }
+   */
+  const pathnameParameterValues: Record<string, string> = {};
```

Comment:
Question for reviewers: Do you think we actually need to include the parameters somewhere in the transaction data?

I am leaning towards no because we also include `originalPathname` in the transaction data which is enough for users to figure out what the parameters were.

Transaction data is currently also not queriable in discover so there wouldn't even be an immediate upside to including it in that regard.

Reply:
Coming back to this even though the PR is closed (😿)

I think we don't include the parameters, it's un-needed bloat on the event JSON.